paths: add a test case for the embeddings dotdot workaround

keegancsmith opened the pull request:
Just ensuring our workaround for a bug in our git show implementation works.

Test Plan: go test

mrnugget reviewed (approved):
Do we know whether this will be slow?

keegancsmith commented:
> Do we know whether this will be slow?

@mrnugget we already include some other patterns by default. I added a benchmark (see code for results) and it takes the same amount of time as a `*.md` like pattern. On my machine that is about 150ns per match. This is insignificantly small compared to calling the embeddings API.

cc @cbart @eseliger I fixed a panic for empty paths and added some benchmarks you might be interested in.
